Framework/DataObjects/inc/MantidDataObjects/VectorColumn.h

```diff
@@ -122,6 +123,59 @@ template <class Type> class DLLExport VectorColumn : public API::Column {
     throw std::runtime_error("VectorColumn is not assignable from double.");
   }
 
+  /// Reference to the data.
+  const std::vector<std::vector<Type>> &data() const { return m_data; }
+
+  bool equals(Column *otherColumn, double tolerance) const override {
+    auto convertedOtherColumn = dynamic_cast<VectorColumn *>(otherColumn);
```

Comment:
Maybe the metadata/type stuff could be in a protected and not private method in the base to avoid duplicating these checks everywhere?